test/new-e2e/pkg/e2e/suite.go

```diff
@@ -223,11 +223,15 @@ func (bs *BaseSuite[Env]) IsDevMode() bool {
 }
 
 func (bs *BaseSuite[Env]) init(options []SuiteOption, self Suite[Env]) {
+	if _, err := runner.GetProfile().ParamStore().GetBoolWithDefault(parameters.DevMode, false); err == nil {
```

Comment:
Why not reading the value? If `E2E_DEV_MODE` is set to false, you should not add `DevMode`.
If `err != nil` you should at least log an error.